### PR TITLE
update readme for recent changes

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

- Removed the "tested file paths" sections as I don't think it's very illuminating any more.
- Updated some notes on file formats (we do have a test against a non-aipy derived miriad file).
- Changed language around uvh5 file format to say it's in beta not alpha (do you agree @plaplant?). 
- Removed known issues items that have been fixed.
- Also fixed some issues with RTD build

## Motivation and Context

The readme hasn't kept up with recent development and we're planning a major release soon, so that should be fixed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Documentation change checklist:
- [ ] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] If this is a significant change to the readme or other docs, I have checked that they are rendered properly on ReadTheDocs. (you may need help to get this branch to build on RTD, just ask!)

### CODE REVIEW

**Comment by bhazelton** [comment]
@dannyjacobs can you review this?

**Comment by mkolopanis** [comment]
Do we want to mention the `version.py` discussion in #590 in the known issue/ development section. Not really API related but definitely something that's been mentioned a few times with other RASG/HERA-Team repos which use a derivative of `pyuvdata.version`

**Comment by bhazelton** [comment]
@mkolopanis I added a line to address this, not sure if I captured it well though. Do you want to take a look?

File: README.md [hunk 1]
```diff
@@ -18,137 +21,163 @@ The three main goals are:
...
+see the [phasing memo](docs/references/phasing.pdf) for more details).
 * UVData: Multiple spectral windows or multiple sources are not currently supported
-* UVData: Add testing against miriad package (currently only tested against aipy derived files)
-* UVData: Concatenation of multiple datasets is somewhat slow.  (see issue #406 for proposed fix)
 * UVData: add support for writing CASA measurement sets
-* UVBeam: support reading/writing/combining standard HEALPix FITs files (individual files per frequency, polarization and E-field vector)
-* UVBeam: support phased-array antenna beams.
+* UVBeam: support phased-array antenna beams (e.g. MWA beams).
+* UVCal/UVData: method to apply calibration to data.
+* package version detection can cause issues with installation directly from the repo for some users.
```
Comment (adampbeardsley) [inline README.md]: let's link to the issue

File: README.md [hunk 2]
```diff
@@ -18,137 +21,163 @@ The three main goals are:
...
+[Bibtex entry](http://adsabs.harvard.edu/cgi-bin/nph-bib_query?bibcode=2017JOSS....2..140H&data_type=BIBTEX&db_key=GEN&nocookieset=1)
 
 # Installation
 For simple installation, the latest stable version is available via conda
-(preferred: ```conda install -c conda-forge pyuvdata```) or pip (```pip install pyuvdata```)
+(preferred: ```conda install -c conda-forge pyuvdata```) or pip (```pip install pyuvdata```).
 
-## Optionally install the development version
-First install the dependencies, see below for package recommendations and
-extra dependencies for HEALPix beams and CASA measurement set functionalities.
```
Comment (adampbeardsley) [inline README.md]: Should we add this back in in light of dependency issue?

File: docs/developer_docs.rst [hunk 3]
```diff
@@ -6,34 +6,34 @@ won't need to interact with.
...
-.. autoclass:: pyuvdata.FHD
+.. autoclass:: pyuvdata.fhd.FHD
   :members:
 
-.. autoclass:: pyuvdata.CALFITS
+.. autoclass:: pyuvdata.calfits.CALFITS
   :members:
 
-.. autoclass:: pyuvdata.BeamFITS
+.. autoclass:: pyuvdata.beamfits.BeamFITS
```
Comment (bhazelton) [inline docs/developer_docs.rst]: need to add: CSTBeam, FHDCal, MS, UVH5

Comment (bhazelton) [reply docs/developer_docs.rst]: also under functions add: aipy_extracts, get_fhd_history, construct_version_info

Comment (bhazelton) [reply docs/developer_docs.rst]: check formatting of get_telescope's docstring

Comment (bhazelton) [reply docs/developer_docs.rst]: clarify distinction between Functions & Utility Functions

File: docs/developer_docs.rst [hunk 4]
```diff
@@ -6,34 +6,34 @@ won't need to interact with.
...
   :members:
 
 .. autoclass:: pyuvdata.Telescope
```
Comment (adampbeardsley) [inline docs/developer_docs.rst]: Adam needs to fix some of the util docstrings (e.g. and_collapse).

**Pull request merged**